evaluate.py

```diff
@@ -55,46 +56,54 @@ def generate(args, g_ema, device, mean_joints, std_joints, entity):
+    if args.dataset == 'mixamo':
+        offsets /= 100  # not needed in humanact  TODO: Why do we need that?
 
+    motion_statics = StaticData(parents=motion_data_raw[0]['parents_with_root'],
+                        offsets=offsets,
+                        names=motion_data_raw[0]['names_with_root'],
+                        character_name=args.character,
+                        n_channels=4,
+                        enable_global_position=args.glob_pos,
+                        enable_foot_contact=args.foot,
+                        rotation_representation=args.rotation_repr)
+
+    dynamic_data = torch.stack([torch.from_numpy(motion[0]) for motion in generated_motion_np]).transpose(1, 2)
+    motions_all = DynamicData(dynamic_data, motion_statics, use_velocity=args.use_velocity)
+    motions_all = motions_all.un_normalise(mean_joints.transpose(0, 2, 1, 3), std_joints.transpose(0, 2, 1, 3))
+
+    generated_motions = convert_motions_to_location(motions_all, args.dataset)
+
+    return generated_motions
 
-def convert_motions_to_location(args, generated_motion_np, edge_rot_dict_general):
-    edge_rot_dict_general['std_tensor'] = edge_rot_dict_general['std_tensor'].cpu()
-    edge_rot_dict_general['mean_tensor'] = edge_rot_dict_general['mean_tensor'].cpu()
-    if args.dataset == 'mixamo':
-        edge_rot_dict_general['offsets_no_root'] /= 100 ## not needed in humanact
 
+def convert_motions_to_location(motions_all, dataset_type):
     generated_motions = []
 
-    # get anim for xyz positions
-    motion_data = un_normalize(generated_motion_np, mean=edge_rot_dict_general['mean'].transpose(0, 2, 1, 3), std=edge_rot_dict_general['std'].transpose(0, 2, 1, 3))
-    anim_dicts, frame_mults, is_sub_motion = edge_rot_dict_from_edge_motion_data(motion_data, type='sample', edge_rot_dict_general = edge_rot_dict_general)
+    for j, motion_all in enumerate(motions_all):
+        anim, names = motion_all.anim_from_static()
 
-    for j, (anim_dict, frame_mult) in enumerate(zip(anim_dicts, frame_mults)):
-        anim, names = anim_from_edge_rot_dict(anim_dict, root_name='Hips')
         # compute global positions using anim
         positions = Animation.positions_global(anim)
 
         # sample joints relevant to 15 joints skeleton
         positions_15joints = positions[:, [7, 6, 15, 16, 17, 10, 11, 12, 0, 23, 24, 25, 19, 20, 21]] # openpose order R then L
         positions_15joints = positions_15joints.transpose(1, 2, 0)
         positions_15joints_oriented = positions_15joints.copy()
-        if args.dataset == 'mixamo':
+        if dataset_type == 'mixamo':
             positions_15joints_oriented = positions_15joints_oriented[:, [0, 2, 1]]
             positions_15joints_oriented[:, 1, :] = -1 * positions_15joints_oriented[:, 1, :]
         generated_motions.append(positions_15joints_oriented)
 
     generated_motions = np.asarray(generated_motions)
     return generated_motions
 
-#endregion
```

Comment:
why removed #endregion? did you remove the matching #startregion?

Reply:
I don't see any startregion, also I don't know this syntax, TO ASK.